### ISSUE TITLE
problems with operator spaces with [align_equals]

### ISSUE THREAD

**Issue by NewFuture** [FriendsOfPHP/PHP-CS-Fixer#2811]
`declare_equal_normalize` dose not work when set the `align_equals = true` 

* The PHP version you are using: v7.0.10
* PHP CS Fixer version you are using: v2.3.2
* configuration :
```
$rules = array(
    'binary_operator_spaces' => array(
        'align_double_arrow' => true,
        'align_equals'       => true,
    ),
    'declare_equal_normalize' => true,
);
```
## A minimal sample of valid PHP code that is not fixed correctly (if applicable).
```php
$a=    1;
$bb=$a;
//some code ...
$other='somthing';
```
result:
```php
$a =    1;
$bb=$a; // no space :(
//some code ...
$other='somthing'; //declare_equal_normalize dose not work :(
```
except:
```php
$a  = 1; //add space and align
$bb = $a;
//some code ...
$other = 'somthing';//add space
```

## Another example:
```php
$a            =  1;
$bb           = $a;
$longlonglong = 2; //then we delete or comment this line
```
like this:
```php
$a            =  1;
$bb           = $a;
//$longlonglong = 2; 
```
when php-cs-fixer result:(nothing is modified)
```php
$a            =  1;
$bb           = $a;
// $longlonglong = 2
```
except:
```php
$a  = 1;
$bb = $a;
// $longlonglong = 2;
```
The order the order of  `declare_equal_normalize` and  `align_equals`  maybe  need to be exchanged.

I think apply the `declare_equal_normalize`  rule at first  and then  apply `align_equals` will fix those problems.

**Comment by keradus** [comment]
`declare_equal_normalize` fixes `declare` statement, in your case there is no such statements.
```
λ php-cs-fixer describe declare_equal_normalize
Description of declare_equal_normalize rule.
Equal sign in declare statement should be surrounded by spaces or not following configuration.

Fixer is configurable using following options:
* space ('none', 'single'): spacing to apply around the equal sign; defaults to 'none'

Fixing examples:
 * Example #1.
   ---------- begin diff ----------
   --- Original
   +++ New
   @@ @@
    <?php
   -declare(ticks =  1);
   +declare(ticks=1);
   ----------- end diff -----------
```

**Comment by NewFuture** [comment]
@keradus thank you     ❤️ . I got it wrong on the  `declare_equal_normalize`. 😿 

But I still  have 2 problems with the `binary_operator_spaces`(`align_double_arrow` and `align_equals` ):

----------
1. The spaces around the  equals(` =`) are not fixed, when configure  `'align_equals' => true`;

example:
```php
//none space before = and two spaces after = .
//it will not  be fixed  when align_equals is ture;
$a=    1;
```
except:

```php
$a = 1;// one space around =
```
Is this a bug?

---------
2. for the algin with `=` and `=>`, the spaces before them will be longer and longer. As the second example:
```php
$a            =  1;
$bb           = $a;
// $longlonglong = 2 

$arr = [
    'a' => 1,
    'b'           =>2,
];
```

except:
```php
$a  =  1;
$bb = $a; // remove the extra spaces before the =
// $longlonglong = 2

$arr = [
    'a' => 1,
    'b' => 2,  // remove the extra spaces before the =>
];
```
Is it possible to fix the code like those？

**Comment by SpacePossum** [comment]
all samples are known and being addressed here;
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2573
please help out if you can

**Comment by NewFuture** [comment]
@SpacePossum it's good features.

But i think this issue(#2811) is a bug.

**Comment by SpacePossum** [comment]
it is, however removing the bug will require refactoring of the fixer as I tried in the PR. With the new feature you  should be able to configure the fixer to do what  you (and I) want, I hope.

**Comment by keradus** [comment]
Closes due to merge of #2573
